rest/route/distro.go

```diff
@@ -10,6 +10,54 @@ import (
 	"github.com/pkg/errors"
 )
 
+////////////////////////////////////////////////////////////////////////
+//
+// GET /rest/v2/distros/{distro_id}
+
+type distroIDGetHandler struct {
+	distroId string
+	sc       data.Connector
+}
+
+func makeGetDistroByID(sc data.Connector) gimlet.RouteHandler {
+	return &distroIDGetHandler{
+		sc: sc,
+	}
+}
+
+func (h *distroIDGetHandler) Factory() gimlet.RouteHandler {
+	return &distroIDGetHandler{
+		sc: h.sc,
+	}
+}
+
+// ParseAndValidate fetches the distroId from the http request.
+func (h *distroIDGetHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.distroId = gimlet.GetVars(r)["distro_id"]
+
+	if h.distroId == "" {
+		return errors.New("request data incomplete")
+	}
+
+	return nil
+}
+
+// Execute calls the data FindDistroById function and returns the distro
+// from the provider.
+func (h *distroIDGetHandler) Run(ctx context.Context) gimlet.Responder {
+	foundDistro, err := h.sc.FindDistroById(h.distroId)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "Database error"))
+	}
+
+	distroModel := &model.APIDistro{}
+	if err = distroModel.BuildFromService(*foundDistro); err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "API model error"))
```

Comment:
we probably need to call the super user redaction code here in your next PR too?